src/error.rs

```diff
@@ -9,83 +9,61 @@ use std::str::FromStr;
 use url::ParseError;
 
 /// An error occurred while attempting to establish a session for a new `Client`.
-#[derive(Debug)]
+#[derive(Debug, thiserror::Error)]
 pub enum NewSessionError {
     /// The given WebDriver URL is invalid.
-    BadWebdriverUrl(ParseError),
+    #[error("webdriver url is invalid: {0}")]
+    BadWebdriverUrl(#[from] ParseError),
```

Comment:
Let's use `#[source]` instead of `#[from]` for any of these we don't already have an explicit `impl From` for. It's not always the case that a public `impl From` makes sense, _especially_ not a single one.